### PR TITLE
docs: replace type mixed in DB Utils

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
See #6310

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Database/BaseUtils.php [hunk 1]
```diff
@@ -260,7 +260,8 @@ public function getXMLFromResult(ResultInterface $query, array $params = []): st
      *
      * @param array|string $params
      *
-     * @return mixed
+     * @return false|string|void
+     * @phpstan-return false|string|never
```
Comment (paulbalandan) [inline system/Database/BaseUtils.php:264]: Can be changed to never for native return?

File: system/Database/MySQLi/Utils.php [hunk 2]
```diff
@@ -36,7 +36,8 @@ class Utils extends BaseUtils
     /**
      * Platform dependent version of the backup function.
      *
-     * @return mixed
+     * @return void
+     * @phpstan-return never
```
Comment (paulbalandan) [inline system/Database/MySQLi/Utils.php:40]: Can be never for native return

File: system/Database/BaseUtils.php (hunk 1 above)
Comment (kenjis) [reply system/Database/BaseUtils.php:264]: Do you mean `@return false|never|string`?

Comment (kenjis) [reply system/Database/BaseUtils.php]: Done.

File: system/Database/MySQLi/Utils.php (hunk 2 above)
Comment (kenjis) [reply system/Database/MySQLi/Utils.php]: Done.

**Comment by kenjis** [comment]
Can someone review?

**Review by paulbalandan — APPROVED** [review]
Sorry, i did not receive notification on the changes.

**Comment by kenjis** [comment]
No problem. Thank you for approval.

**Pull request merged**